Core_OverlayMods/Clothes/KoiClothesOverlayController.cs

```diff
@@ -118,6 +139,72 @@ public static bool IsMaskKind(string clothesId)
             return false;
 #endif
         }
+
+        public static string GetColormaskId(string clothesId, int kindId, int subKindId = -1)
+        {
+            return $"{clothesId}_Colormask_{kindId}_{subKindId}";
+        }
+
+        public static bool IsColormask(string clothesId)
+        {
+            return clothesId.Contains("_Colormask_");
+        }
+
+        public static int[] GetKindIdsFromColormask(string clothesId)
+        {
+            if (!IsColormask(clothesId))
+                return null;
+
+            var match = Regex.Match(clothesId, @"_Colormask_(\d)_(-?\d)");
+            Int32.TryParse(match.Groups[1].Value, out var kindId);
+            Int32.TryParse(match.Groups[2].Value, out var subKindId);
+
+            return new[] { kindId, subKindId };
+        }
```

Comment:
With some changes to the parsing it might be possible to combine these two? Have both IDs as 2 out parameters. Would need to not be regex though. Maybe reshuffle the format to `ColorMask_ID1_ID2_name` and if ID2 does not exist just keep it blank `ColorMask_ID1__name`. That way simple and fast index checks would be possible.